pal/input/native/touch-input.ts

```diff
@@ -25,63 +25,132 @@
 import { screenAdapter } from 'pal/screen-adapter';
 import { Size, Vec2 } from '../../../cocos/core/math';
 import { EventTarget } from '../../../cocos/core/event';
-import { EventTouch, Touch } from '../../../cocos/input/types';
+import { EventTouch, Touch as CCTouch } from '../../../cocos/input/types';
 import { touchManager } from '../touch-manager';
-import { macro } from '../../../cocos/core/platform/macro';
 import { InputEventType } from '../../../cocos/input/types/event-enum';
 
 export type TouchCallback = (res: EventTouch) => void;
 
 declare const jsb: any;
 
+class TouchEventElement {
+    type: InputEventType | null = null;
+    changedTouches: Touch[] = [];
+    windowId: number = 0;
+}
+
+class TouchEventCache {
+    private _events: TouchEventElement[] = [];
+    private _length = 0;
+
+    push (eventType: InputEventType, changedTouches: Touch[], windowId: number): void {
+        const events = this._events;
+        const index = this._length;
+        if (index >= events.length) {
+            events.push(new TouchEventElement());
+        }
+        const e = events[index];
+        const cachedTouches = e.changedTouches;
+        e.type = eventType;
+        cachedTouches.length = changedTouches.length;
+        e.windowId = windowId;
+
+        for (let i = 0, len = changedTouches.length; i < len; ++i) {
+            const src = changedTouches[i];
+            let dst = cachedTouches[i] as any;
+            if (!dst) {
+                (cachedTouches[i] as any) = dst = {
+                    identifier: 0,
+                    clientX: 0,
+                    clientY: 0,
+                    pageX: 0,
+                    pageY: 0,
+                };
```

Comment:
Can assign the value with src.

Reply:
Updated.

Use `Object.assign` to copy data from native.